### PR TITLE
GET /plugins/ returns both installed and available plugins

### PR DESCRIPTION
**Pull request by valentimarco**
# Description

2 new function to retrieve from registry all external plugins available and to install plugins by url.

For now the implementation of "retrieve from registry" bricks plugin page on admin!

Related to issue #300 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: core/cat/routes/plugins.py [hunk 1]
```diff
@@ -182,3 +189,47 @@ async def delete_plugin(plugin_id: str, request: Request) -> Dict:
...
     }
+
+async def get_registry_list():
+    response = httpx.get("http://192.168.1.120:8000/plugins?page=1&page_size=7000")
+    if response.status_code == 200:
+        return response.json()["plugins"]
+    else: 
+        return []
+
+@router.post("/upload/wa")
```
Comment (pieroit) [inline core/cat/routes/plugins.py:200]: Endpoint is `/upload/registry`

File: core/cat/routes/plugins.py [hunk 2]
```diff
@@ -182,3 +189,47 @@ async def delete_plugin(plugin_id: str, request: Request) -> Dict:
...
+async def download_plugin_from_registry(request: Request,background_tasks: BackgroundTasks,url_repo: Dict = Body(example={"url": "https://example.com/file.zip"})):
```
Comment (pieroit) [inline core/cat/routes/plugins.py:201]: Example should be the URL of a plugin on github

File: core/cat/routes/plugins.py [hunk 3]
```diff
@@ -182,3 +189,47 @@ async def delete_plugin(plugin_id: str, request: Request) -> Dict:
...
+    url_path.reverse()
+    if "github" in url.netloc:
+        print(url_path)
+        plugin_name = str(url_path[2]) + ".zip"
+    else:
+        plugin_name = str(url_path[0])
+    
+    print(plugin_name)
+    with requests.get(url_repo["url"], stream=True) as response:
+        response.raise_for_status()
```
Comment (pieroit) [inline core/cat/routes/plugins.py:216]: Use `HTTPException` from fastAPI (consistent across al endpoints)

**Review by pieroit — CHANGES_REQUESTED** [review]

**Comment by pieroit** [comment]
Why closed? Just put the PR as a draft until you finish :)

**Comment by valentimarco** [comment]
i didn't nothing, i synced the fork. Maybe that?

**Comment by pieroit** [comment]
I see zero commits on the PR
Let me know, you can reopen this or maie a new one

**Comment by valentimarco** [comment]
> I see zero commits on the PR Let me know, you can reopen this or create a new one

Re open the PR

**Comment by pieroit** [comment]
Thanks @valentimarco congrats on your merged PR :D

Let's wait a little before testing this code because the responsibility of finding and caching the .zip file of plugins may rest on [this repo](https://github.com/cheshire-cat-ai/awesome-plugins-backend).

Let's discuss this further with @EugenioPetulla @nicola-corbellini

**Pull request merged**